### ISSUE TITLE
[BUG] RustServer - Offline (sleeping) players spawn on beach when killed

### ISSUE THREAD

**Issue by rogderdodger** [GameServerManagers/LinuxGSM#3793]
## Basic info

* **Distro:** [Ubuntu 18.04]
* **Game:** [Rust]
* **LinuxGSM version:** [latest]

## Further Information

Hello everyone, i think I have identified a problem with the Rustserver on Linux GSM. 

The problem is a weird one, when players on the server die while offline (sleeping) and log back into the game, they are not presented with the default 'death screen'.
I've been troubleshooting this for a while and speaking to other server owners who have also been asking for help with the issue, so far i've identified that they all use Linux GSM to host rust. 

So far i have tested many things. 

Starting a new rust server instance with Linux LGSM (no oxide or mods) 
Removing oxide and all mods from my existing servers
Reinstalling my existing server 

I have also verified a few other things. 

This does not happen on windows 
This does not happen with a fresh install just using steamcmd on linux 
Adding Oxide to the manual steamcmd install still works fine 
This does not happen when using pterodactyl on linux 

It only seems to happen when hosting rust using Linux GSM. 

This causes a lot of problems for many server owners using Linux GSM and i am really hopeful someone can help fix the issue.
## To Reproduce

Steps to reproduce the behaviour:
1. Create a vanilla rust server 
2. Load into the server with rust 
3. die offline (i do this by throwing a c4 at my feet and disconnecting) 
4. reconnect
5. You will spawn on the beach without death screen

## Expected behaviour

with a fresh manual steamcmd install - following the same steps you will see the death screen instead of just spawning on the beach

**Comment by rogderdodger** [comment]
Another test is that running a basic start.sh script within the Serverfiles folder using the same startup parameters as in the rustserver.config file also does not have this problem. 

Stopping that server and using ./rustserver start however, the problem persists

**Comment by rogderdodger** [comment]
After more investigation i have found that the following change to the start parameters in rustserver.cfg resolves the issue. 

startparameters="-batchmode -nographics +app.listenip ${ip} +app.port ${appport} +server.ip ${ip} +server.port ${port} +server.tickrate ${tickrate} +server.hostname \"${servername}\" +server.identity \"${selfname}\" +server.level \"${serverlevel}\" +server.seed ${seed} +server.salt ${salt} +server.maxplayers ${maxplayers} +server.worldsize ${worldsize} +server.saveinterval ${saveinterval} +rcon.web ${rconweb} +rcon.ip ${ip} +rcon.port ${rconport} +rcon.password \"${rconpassword}\" -logfile"


(Added -nographics)
(removed - server.gamemode "Vanilla")

**Comment by pwjmc** [comment]
> 

Is this stull working for you?

**Comment by rogderdodger** [comment]
> 

Yes this still works and resolves the issue

**Comment by nimdaaaa** [comment]
Hey Dude,
find this over google, had to deal a long time with this problem.
So you just removed gamemode vanilla and it works ?
Or whats this with the $ before each start parameter.
Greetings

**Comment by rogderdodger** [comment]
Hey yeah just remove the game mode parameter and it should solve the problem, I changed a couple of things and LGSM runs perfect now, full instructions are in my post.

The $ symbols were already in my config file, if yours doesn’t have them don’t use them

**Comment by nimdaaaa** [comment]
Dude , love it...... will try directly. 
Fucked me up since 5-6 months now.....
Thought it was plugin related , but with so much plugins hard to find out.
Will let you know ;)

**Comment by pwjmc** [comment]
I can confirm this fix works perfectly. I've been using it for 2 months across 7 servers.  You're a genius rogderdodger, I thought I was going crazy.

**Comment by rogderdodger** [comment]
great - yeah this drove me crazy for months too. its such a simple fix but took a long time and a lot of trying different things to narrow it down. 

I was a very happy server owner when i first saw that death screen! 

All the best

**Comment by nimdaaaa** [comment]
Dude , it fucking works........
Normaly i got every fucking failure soveld by myself...... 
just that one leaves me with so much headache.
Dont know what to say , i tried to disable 100 plugins , googled my ass off.......
and that little fucking vailla server line fucked it up !

Dude loveit.

but got interested in that youre running 7 servers.... Youre active in Rustedit too ?

Greetings

**Comment by TerpenBoy** [comment]
Hey upon deleting server.gamemode line from the server config does it force wipe or delete things out of players inventory want to try this but waiting till end of wipe. Would be nice to try it already though.

**Comment by rogderdodger** [comment]
@TerpenBoy no, it won't wipe or cause you any problems as long as you don't change your seed or worldsize values. I'd suggest if you're unsure to backup your  serverfiles/server/rustserver directory just in case but you should be fine to do it whenever. Good luck!

**Comment by Anfernee12345** [comment]
It also works when your server is built on windows. I had the same problem, and removed the gamemode and it worked.
Thanks alot for the solution.
